wpilibNewCommands/src/test/java/edu/wpi/first/wpilibj2/command/CancelFromInitializeTest.java

```diff
@@ -36,14 +38,49 @@ public void initialize() {
 
       assertDoesNotThrow(
           () -> {
-            scheduler.schedule(true, selfCancels);
+            scheduler.schedule(interruptible, selfCancels);
             scheduler.run();
-            scheduler.schedule(true, other);
+            // interruptibility of new arrival isn't checked
+            scheduler.schedule(other);
           });
       assertFalse(scheduler.isScheduled(selfCancels));
       assertTrue(scheduler.isScheduled(other));
       scheduler.run();
       assertTrue(hasOtherRun.get());
     }
   }
+
+  @ValueSource(booleans = {true, false})
+  @ParameterizedTest
+  void defaultCommand(boolean interruptible) {
+    try (CommandScheduler scheduler = new CommandScheduler()) {
+      AtomicBoolean hasOtherRun = new AtomicBoolean();
+      Subsystem requirement = new SubsystemBase() {};
+      Command selfCancels =
+              new CommandBase() {
+                {
+                  addRequirements(requirement);
+                }
+
+                @Override
+                public void initialize() {
+                  scheduler.cancel(this);
+                }
+              };
+      Command other = new RunCommand(() -> hasOtherRun.set(true), requirement);
+      scheduler.setDefaultCommand(requirement, other);
+
+      assertDoesNotThrow(
+              () -> {
+                scheduler.schedule(interruptible, selfCancels);
+                scheduler.run();
+              });
+      scheduler.run();
+      assertFalse(scheduler.isScheduled(selfCancels));
+      assertTrue(scheduler.isScheduled(other));
+      scheduler.run();
+      assertTrue(hasOtherRun.get());
+    }
+  }
```

Comment:
Nice test-driven development.  I have no reason to believe that there are problems elsewhere, but if we truly want to ensure that `cancel` is safe to call without restriction, here are some other contexts we ought to check
* Within `execute`, `isFinished` and `end` (!) of the command.
* Within `execute` of another command (so inside the run loop).
* Within a subsystem `periodic` (so outside the run loop).

In all cases, we should probably consider both the reschedule case and the default command case.